File: Python3/601 - 800/645 Set Mismatch.py
# 64ms, 83.33%
class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res, is_exist = [], [False] * (len(nums) + 1)
        for n in nums:
            if is_exist[n]:
                res.append(n)
            else:
                is_exist[n] = True
        for i, e in enumerate(is_exist[1:], 1):
            if not e:
                res.append(i)
                break
        return res
    

File: 601 - 800/645 Set Mismatch.cc
// 8ms, 100.0%
static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution
{
  public:
    vector<int> findErrorNums(const vector<int> &nums)
    {
        vector<int> res;
        bool *isExist = new bool[nums.size() + 1]{};
        for (const auto x : nums)
            if (isExist[x]) {
                res.push_back(x);
            } else {
                isExist[x] = true;
            }
        for (int i = 1, n = nums.size(); i <= n; ++i)
            if (!isExist[i]) {
                res.push_back(i);
                break;
            }
        return res;
    }
};
